rs/nns/governance/src/proposals/mod.rs

```diff
@@ -1,4 +1,52 @@
+use crate::pb::v1::{governance_error::ErrorType, GovernanceError, Topic};
+use ic_base_types::CanisterId;
+use ic_nns_constants::{
+    BITCOIN_MAINNET_CANISTER_ID, BITCOIN_TESTNET_CANISTER_ID, CYCLES_LEDGER_CANISTER_ID,
+    CYCLES_LEDGER_INDEX_CANISTER_ID, CYCLES_MINTING_CANISTER_ID, EXCHANGE_RATE_CANISTER_ID,
+    GENESIS_TOKEN_CANISTER_ID, GOVERNANCE_CANISTER_ID, ICP_LEDGER_ARCHIVE_1_CANISTER_ID,
+    ICP_LEDGER_ARCHIVE_CANISTER_ID, LEDGER_CANISTER_ID, LEDGER_INDEX_CANISTER_ID,
+    LIFELINE_CANISTER_ID, REGISTRY_CANISTER_ID, ROOT_CANISTER_ID, SUBNET_RENTAL_CANISTER_ID,
+};
+
 pub mod call_canister;
 pub mod create_service_nervous_system;
 pub mod install_code;
 pub mod proposal_submission;
+pub mod stop_or_start_canister;
+
+const PROTOCOL_CANISTER_IDS: [&CanisterId; 16] = [
+    &REGISTRY_CANISTER_ID,
+    &GOVERNANCE_CANISTER_ID,
+    &LEDGER_CANISTER_ID,
+    &ROOT_CANISTER_ID,
```

Comment:
Should root be here? I don't even know if root can stop itself, but if it can, then wouldn't there be no way to restart it?

Reply:
Nice catch! It's enforced at root, so proposals would pass validation (and get created) and fail at execution. Now that we have the structured data at Governance I think we should enforce it at validation. Will update this PR.

Reply:
Done.